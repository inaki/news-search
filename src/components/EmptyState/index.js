// EmptyState Component
import React from 'react';
import styled from 'styled-components';

const StyledEmptyState = styled.div`
    width: 100wv;
    height: 100vh;
    font-size: 1.7rem;
    color: darkgray;
    text-align: center;
    margin-top: 20%;
`;

const EmptyState = ({text = 'No Results...'}) => {
    return (
        <StyledEmptyState>
            {text}
        </StyledEmptyState>
    );
}

export default EmptyState;