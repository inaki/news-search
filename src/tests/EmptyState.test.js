import React from 'react';
import { shallow } from 'enzyme';
import EmptyState from '../components/EmptyState';

describe('EmptyState Component', () => {
    let app;
    beforeEach(() => {
        app = shallow(<EmptyState />)
    });
    
    it('reders properly', () => {
        expect(app).toMatchSnapshot();
    });

    it('render text for empty state', () => {
        expect(app.text()).toEqual('No Results...');
    });

});
