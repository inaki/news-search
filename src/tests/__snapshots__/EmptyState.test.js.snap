// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Component reders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyState />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "No Results...",
    },
    "ref": null,
    "rendered": "No Results...",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    width: 100wv;
    height: 100vh;
    font-size: 1.7rem;
    color: darkgray;
    text-align: center;
    margin-top: 20%;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "No Results...",
      },
      "ref": null,
      "rendered": "No Results...",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    width: 100wv;
    height: 100vh;
    font-size: 1.7rem;
    color: darkgray;
    text-align: center;
    margin-top: 20%;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
